--- a/Modules/grpmodule.c
+++ b/Modules/grpmodule.c
@@ -258,18 +258,7 @@
 {
     PyObject *d;
     struct group *p;
     if ((d = PyList_New(0)) == NULL)
         return NULL;
-    setgrent();
-    while ((p = getgrent()) != NULL) {
-        PyObject *v = mkgrent(module, p);
-        if (v == NULL || PyList_Append(d, v) != 0) {
-            Py_XDECREF(v);
-            Py_DECREF(d);
-            endgrent();
-            return NULL;
-        }
-        Py_DECREF(v);
-    }
-    endgrent();
     return d;
 }
